docs/content/migration.md

```diff
@@ -0,0 +1,545 @@
+| toolbar            |   [Docs][66] |   [Docs][7]  |                                                        |
+| tooltip            |   [Docs][67] |   [Docs][18] |                                                        |
+| truncate           |   [Docs][103]|   -          | Not implemented                                        |
+| virtual-repeat     |   [Docs][68] |   [Docs][40] | `cdk-virtual-scroll-viewport` and `*cdkVirtualFor`     |
+| whiteframe         |   [Docs][104]|  [Guide][105]| Based on classes and mixins                            |
+
+### Services
+
+| Service          | Old          | New          | Special Notes                                          |
+|------------------|--------------|--------------|--------------------------------------------------------|
+| $mdAriaProvider  |   [Docs][124]|   -          | Not implemented                                        |
+| $mdBottomSheet   |   [Docs][42] |   [Docs][38] | `MatBottomSheet`                                       |
+| $mdColors        |   [Docs][125]|   -          | [`mat-color` mixin][74] supports static theme color lookups.|
+| $mdCompiler      |   [Docs][126]|   -          | Not implemented                                        |
+| $mdCompilerProvider| [Docs][127]|   -          | Not implemented                                        |
+| $mdDateLocaleProvider|[Docs][128]|  [Docs][129]| `MomentDateAdapter`                                    |
+| $mdDialog        |   [Docs][48] |   [Docs][22] | `MatDialog`                                            |
+| $mdGestureProvider|  [Docs][130]|   -          | Not implemented                                        |
+| $mdIcon          |   [Docs][131]|   [Docs][132]| `svgIcon`                                              |
+| $mdIconProvider  |   [Docs][133]|   [Docs][134]| `MatIconRegistry`                                      |
+| $mdInkRipple     |   [Docs][58] |   [Docs][19] |                                                        |
+| $mdInkRippleProvider|[Docs][135]|   [Docs][136]| `MAT_RIPPLE_GLOBAL_OPTIONS`                            |
+| $mdInteraction   |   [Docs][137]|   -          | Not implemented                                        |
+| $mdLiveAnnouncer |   [Docs][111]|   [Docs][112]| CDK `LiveAnnouncer`                                    |
+| $mdMedia         |   [Docs][137]|   [Docs][138]| CDK `BreakpointObserver.isMatched()` or Flex Layout's [MediaObserver service][119]|
+| $mdPanel         |   [Docs][109]|   [Docs][110]| CDK `Overlay`                                          |
+| $mdPanelProvider |   [Docs][139]|   -          | Not implemented                                        |
+| $mdProgressCircularProvider|[Docs][140]|-      | Not implemented                                        |
+| $mdSidenav       |   [Docs][141]|   -          | Not implemented                                        |
```

Comment:
I wouldn't say this is unimplemented so much as no longer necessary with the `MatSidenav` component

Reply:
This is just a convenience to allow something like `$mdSidenav('right').toggle()` and `$mdSidenav('right').toggle()` for use from components without the need to have an actual reference to the specific `md-sidenav` instance.

I wasn't able to find any service that provides this convenience in Angular Material. You need to manually grab a reference to the instance and then stash it in an application-wide service for later use.

Some of the features implemented by these services are still possible via custom code in Angular Material. Should I add a section here that explains how to accomplish some of these use cases in Angular Material?

Reply:
I think in Angular the idiomatic way to do this is by accessing the template reference variables or through `ViewChild`. 

I would consider it an anti-pattern to have a global registry of sidenav's in Angular, so maybe saying this is not implemented is wrong. It's just implemented in a Angular idiomatic way.